src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(String.format("Report for period %s - %s", dateFrom, dateTo))
+                .append(System.lineSeparator());
+        int[] salaryInfo = new int[names.length];
+        for (int i = 0; i < names.length; i++) {
+            for (String checkedData : data) {
+                String[] info = checkedData.split(" ");
+                if (names[i].equals(info[1])) {
+                    try {
+                        LocalDate dateAfter = LocalDate.parse(dateFrom, formatter);
+                        LocalDate dateBefore = LocalDate.parse(dateTo, formatter);
```

Comment:
```suggestion
                        LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
                        LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
```
Create these vars before loops.